framework/src/org/checkerframework/framework/flow/CFAbstractStore.java

```diff
@@ -369,8 +369,11 @@ public void insertValue(FlowExpressions.Receiver r, /*@Nullable*/ V value) {
      * @return true if fieldAcc is an update of a monotonic qualifier to its target qualifier.
      * (e.g. @MonotonicNonNull to @NonNull)
      */
-    private boolean isMonotonicUpdate(FieldAccess fieldAcc, V value) {
+    protected boolean isMonotonicUpdate(FieldAccess fieldAcc, V value) {
         boolean isMonotonic = false;
+        // TODO: this check for !sequentialSemantics is an optimization that breaks the contract of the method,
+        // since the method name and documentation say nothing about sequential semantics.
```

Comment:
Is there any reason why there is this dependency? Why can't you have monotonic updates in sequential mode?